src/main/java/de/tum/in/www1/artemis/domain/Achievement.java

```diff
@@ -0,0 +1,147 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.enumeration.AchievementRank;
+import de.tum.in.www1.artemis.domain.enumeration.AchievementType;
+
+@Entity
+@Table(name = "achievement")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class Achievement implements Serializable {
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
```

Comment:
We introduced a new superclass for domain classes a few days ago that comes with the benefits of
- already implementing `Serializable`
- already having an `id` attribute including `getId()` and `setId(Long)`
- already overriding `Object.hashCode()` and `Object.equals(Object)` correctly, based on the id.

I think it would be a good idea (and with the tests of #2155 almost necessary) to use `DomainObject` here.
Then there are also fewer problems to deal with.

```suggestion
public class Achievement extends DomainObject {
```

Reply:
Awesome, thank you for the hint and explanation 👍
Applied the changes as suggested